Backpack/BarChart/Classes/BarChartCollectionView.swift

```diff
@@ -0,0 +1,149 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers
+@objc(BPKBarChartCollectionView)
+public final class BPKBarChartCollectionView: UICollectionView {
+    
+    /// The BPKBarChartDataSource which will infom how the bar chart should be rendered
+    public var barChartDataSource: BPKBarChartCollectionViewDataSource?
+    
+    /// The BPKBarChartDelegate which can be used to respond to interaction with the bar chart
+    public weak var barChartDelegate: BPKBarChartCollectionViewDelegate?
+
+    /// The selected indexPath
+    var selectedIndexPath: IndexPath? {
+        didSet {
+            updateSelectedMarkerPosition()
+        }
+    }
+    
+    fileprivate static let cellIdentifier: String = "BPKBarChartCollectionView_CellIdentifier"
+    fileprivate var selectedMarkerBottomConstraint: NSLayoutConstraint?
+
+    public override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
+        super.init(frame: frame, collectionViewLayout: layout)
+        setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupViews()
+    }
+    
+    func setupViews() {
+        backgroundView = UIView() // <-- This is a bit hacky!
```

Comment:
I'll say, this view is never added to the view hierarchy and it's not positioned. I think we need to find a better method than this

Reply:
Yeah I agree. It's strange that this works. Maybe the green bar should be rendered by the parent component 🤷

Reply:
I think we should explore constraining it to the collection view some more

Reply:
I would try [`frameLayoutGuide`](https://developer.apple.com/documentation/uikit/uiscrollview/2865772-framelayoutguide)

Reply:
YOU ARE A GENIUS!!!!!

Reply:
@k0nserv you're too smart to be here, get out. 😆